lib/validator.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+class Validator {
+  isString(input) {
+    return typeof input === 'string';
+  }
+
+  isNumber(input) {
+    return typeof input === 'number';
+  }
+
+  isArray(input) {
+    return Array.isArray(input);
+  }
+
+  isObject(input) {
+    return typeof input === 'object';
+  }
+
+  isFunction(input) {
+    return typeof input === 'function';
+  }
+
+  isBoolean(input) {
+    return typeof input === 'boolean';
+  }
+
+  isCorrectType(value, type) {
+    switch (type) {
+      case 'string':
+        return this.isString(value);
+      case 'number':
+        return this.isNumber(value);
+      case 'array':
+        return this.isArray(value);
+      case 'object':
+        return this.isObject(value);
+      case 'function':
+        return this.isFunction(value);
+      case 'boolean':
+        return this.isBoolean(value);
+    }
+  }
+
+  isTruthy(value) {
+    return !!value || value === 0 || value === false;
+  }
+
+  isValid(input, model) {
+    for (let fieldName in model.fields) {
+      let field = model.fields[fieldName];
```

Comment:
when you add the constructor line 50 and 51 might need to change to correspond with constructor, for example to this.schema.fields[field name]